server/src/main/java/org/opensearch/action/admin/indices/settings/get/TransportGetSettingsAction.java

```diff
@@ -121,7 +121,12 @@ protected void clusterManagerOperation(GetSettingsRequest request, ClusterState
                 continue;
             }
 
-            Settings indexSettings = settingsFilter.filter(indexMetadata.getSettings());
+            Settings indexSettingsWithRoutingShards = Settings.builder()
+                .put(indexMetadata.getSettings())
+                .put(IndexMetadata.INDEX_NUMBER_OF_ROUTING_SHARDS_SETTING.getKey(), indexMetadata.getRoutingNumShards())
```

Comment:
I see there're some unit test code for `TransportGetSettingsAction` in this file:https://github.com/opensearch-project/OpenSearch/blob/f8e886507950d13e7e7d3f7d9bd44241e5ff83b7/server/src/test/java/org/opensearch/action/admin/indices/settings/get/GetSettingsActionTests.java#L74
, please also add some unit test cases for the code change in this PR, thanks!

Reply:
Added a test case for the code change in `GetSettingsActionTests`